File: my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class SmartphoneNode : public rclcpp::Node
{
public:
    SmartphoneNode() : Node("smartphone")
    {
        // Initialize subscriber (topic_name, queue_size, callback).
        // we need to bind the callback (&Sma...News) the smartphone node (this). Then we need to place
        // a placeholder_1 because we have one param in the callback (const ... msg)
        subscriber_ = this->create_subscription<std_msgs::msg::String>(
            "robot_news", 10,
            std::bind(&SmartphoneNode::callbackRobotNews, this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), "Smartphone has been started.");
    }

private:
    // callback that is called by the subs. We need a shared pointer with const type.
    void callbackRobotNews(const std_msgs::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
    }
    // Create subscriber (also this a shared pointer)
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscriber_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SmartphoneNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}

File: my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

# Inherit from Node obj
class SmartphoneNode(Node):
    def __init__(self):
        super().__init__("smartphone")
        # create_subscription(data_type_of_topic, topic_name, Callback, queue_size)
        # The callback is created only AFTER the spin. In the first run only the node of the creation
        # of the node is done
        self.subscriber_ = self.create_subscription(String, "robot_news", self.callback_robot_news, 10)
        self.get_logger().info("smartphone has been started.")

    def callback_robot_news(self, msg):
        self.get_logger().info(msg.data)

def main(args=None):
    rclpy.init(args=args)
    node = SmartphoneNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__=="__main__":
    main()